src/main/scala/com/gu/subscriptions/cas/service/SubscriptionService.scala

```diff
@@ -31,8 +31,9 @@ class SubscriptionService(zuoraService: ZuoraService,
       if (allPlanBenefits.contains(Digipack)) {
         for {
           account <- zuoraService.getAccount(subscription.accountId)
-          contact <- zuoraService.getContact(account.billToId)
-        } yield contact.matchesPassword(password)
+          billToContactMatches <- zuoraService.getContact(account.billToId).map(_.matchesPassword(password))
+          valid <- if (billToContactMatches) Future(true) else zuoraService.getContact(account.soldToId).map(_.matchesPassword(password))
```

Comment:
`Future.successful(true)` would have done here - I think this came up before and I can't remember what we worked out, but it was about whether this would spin off a separate promise and complete it later